### PR TITLE
TRD: Use of a signal array structure instead of extra time bin for keeping MC label index

### PR DESCRIPTION
**Pull request by jolopezl**
In preparation for adding pileup to the TRD digitizer.

- Use of a signal array structure instead of extra time bin for keeping MC label index.
- Cleaning of Digit class, making it POD. Signal handler methods moved to the Digitizer, since they are not used elsewhere for now.

### CODE REVIEW

File: Detectors/TRD/base/include/TRDBase/Digit.h [hunk 1]
```diff
@@ -50,61 +45,19 @@ class Digit
...
   void setRow(int row) { mRow = row; }
   void setPad(int pad) { mPad = pad; }
-  void setADC(ArrayADC_t adc) { mADC = adc; }
+  void setADC(ArrayADC adc) { mADC = adc; }
   // Get methods
   int getDetector() const { return mDetector; }
   int getRow() const { return mRow; }
   int getPad() const { return mPad; }
-  ArrayADC_t getADC() const { return mADC; }
+  ArrayADC getADC() const { return mADC; }
```
[the remark this answers is not in the document]
Comment (jolopezl) [reply Detectors/TRD/base/include/TRDBase/Digit.h]: Thanks for pointing that out, changed.

File: Detectors/TRD/base/include/TRDBase/Digit.h [hunk 2]
```diff
@@ -50,61 +45,19 @@ class Digit
...
   void setRow(int row) { mRow = row; }
   void setPad(int pad) { mPad = pad; }
-  void setADC(ArrayADC_t adc) { mADC = adc; }
+  void setADC(ArrayADC adc) { mADC = adc; }
   // Get methods
   int getDetector() const { return mDetector; }
   int getRow() const { return mRow; }
   int getPad() const { return mPad; }
-  ArrayADC_t getADC() const { return mADC; }
+  ArrayADC getADC() const& { return mADC; }
```
Comment (sawenzel) [inline Detectors/TRD/base/include/TRDBase/Digit.h]: It should be `ArrayACD const& getADC() cont {return mADC;}`. Same everywhere else. For instance a few lines up `setADC(ArrayADC const& adc) { mADC = adc; }`

Comment (jolopezl) [reply Detectors/TRD/base/include/TRDBase/Digit.h]: Thanks, fixed now.

**Comment by jolopezl** [comment]
Failing test in build/o2checkcode/o2 unrelated to this PR.

**Pull request merged**